invokeai/backend/model_manager/install.py

```diff
@@ -0,0 +1,758 @@
+
+        cls = ModelInstallJob
+        if re.match(REPO_ID_RE, str(source)):
+            cls = ModelInstallRepoIDJob
+            kwargs = dict(variant=variant)
+        elif re.match(HTTP_RE, str(source)):
+            cls = ModelInstallURLJob
+            kwargs = {}
+        else:
+            raise ValueError(f"'{source}' is not recognized as a local file, directory, repo_id or URL")
+        return cls(
+            source=str(source),
+            destination=Path(self._tmpdir.name),
+            access_token=access_token,
+            priority=priority,
+            **kwargs,
+        )
+
+    def wait_for_installs(self) -> Dict[Union[str, Path, AnyHttpUrl], Optional[str]]:
+        """Pause until all installation jobs have completed."""
+        self._download_queue.join()
+        id_map = self._async_installs
+        self._async_installs = dict()
+        return id_map
+
+    def scan_directory(self, scan_dir: Path, install: bool = False) -> List[str]:  # noqa D102
+        self._cached_model_paths = set([Path(x.path) for x in self._store.all_models()])
+        callback = self._scan_install if install else self._scan_register
+        search = ModelSearch(on_model_found=callback)
+        self._installed = set()
+        search.search(scan_dir)
+        return list(self._installed)
+
+    def hash(self, model_path: Union[Path, str]) -> str:  # noqa D102
+        return FastModelHash.hash(model_path)
+
+    def convert_model(
+        self,
+        key: str,
+        dest_directory: Optional[Path] = None,
+    ) -> ModelConfigBase:
+        """
+        Convert a checkpoint file into a diffusers folder.
+
+        It will delete the cached version ans well as the
+        original checkpoint file if it is in the models directory.
+        :param key: Unique key of model.
+        :dest_directory: Optional place to put converted file. If not specified,
+        will be stored in the `models_dir`.
+
+        This will raise a ValueError unless the model is a checkpoint.
+        This will raise an UnknownModelException if key is unknown.
+        """
+        from .loader import ModelInfo, ModelLoad  # to avoid circular imports
+
+        new_diffusers_path = None
+
+        try:
+            info: ModelConfigBase = self._store.get_model(key)
+
+            if info.model_format != "checkpoint":
+                raise ValueError(f"not a checkpoint format model: {info.name}")
+
+            # We are taking advantage of a side effect of get_model() that converts check points
+            # into cached diffusers directories stored at `path`. It doesn't matter
+            # what submodel type we request here, so we get the smallest.
+            loader = ModelLoad(self._app_config)
+            submodel = {"submodel_type": SubModelType.Scheduler} if info.model_type == ModelType.Main else {}
+            converted_model: ModelInfo = loader.get_model(key, **submodel)
```

Comment:
Without fully groking what's going on here, model converting kinda feels like it should be its own thing and not a implicit behaviour

Reply:
Conversion occurs deep down in Stalker's model loading code and is not easy to call directly without duplicating a lot of edge case handling for VAE overrides, EMA vs non-EMA unets, etc. So this was expedient and safe. At some point we need to tackle the loading code, but not today.

Reply:
Fair.